docs/getting_started.md

```diff
@@ -0,0 +1,262 @@
+
+On **Linux** you can install envsubst by installing the gettext RPM.
+```
+$ apt install gettext
+# or...
+$ yum install gettext
+```
+
+On **MacOS** you can install envsubst with brew:
+```
+$ brew install gettext
+```
+
+NOTE: brew may install envsubst in a path that is not in your PATH environment variable. Ensure
+
+#### clusterctl
+
+A version of `clusterctl` that is built from **this** respository must be installed. If you have a workin Go environment,
+you can use the `clusterctl` make target:
+```
+$ make clusterctl
+```
+
+The `clusterctl` make target installs `clusterctls` in `bin/clusterctl`. Temporarily add this folder to your PATH to use it
+for the rest of this guide:
+```
+$ export PATH=$PATH:$(pwd)/bin
+```
+
+### vSphere Requirements
+
+#### vCenter Credentials
+
+In order for `clusterctl` to bootstrap a management cluster on vSphere, it must be able to connect and authenticate to vCenter.
+Ensure you have credentials to your vCenter server (user, password and server URL).
+
+#### Networking Requirements
+
+**TODO**: this needs to be updated based on akutz's recent changes with generating kubeconfig
+
+### Generating YAML for the Bootstrap Cluster
+
+The bootstrapping process in `clusterctl` requires a few configuration files:
+* **cluster.yaml**: a yaml file defining the cluster resource for your management cluster
+* **machine.yaml**: a yaml file defining the machine resource for the initial control plane node of your management cluster
+* **provider-components.yaml**: a yaml file which adds all the Cluster API and CAPV-specific resources to your management cluster
+* **addons.yaml**: a yaml file indicating any additional addons you want on your management cluster (e.g. CNI plugin)
+
+The project Makefile provides a convience make target to generate the yaml files. Before attempting to generate the above yaml file,
+the following environment variables **must** be set based on your vSphere environment:
+
+```bash
+# vCenter config/credentials
+export VSPHERE_SERVER=10.0.0.1                # the vCenter server IP or URL
+export VSPHERE_USER=viadmin@vmware.local      # the vCenter user to login with
+export VSPHERE_PASSWORD=some-secure-password  # the vCenter password to login with
+
+# vSphere deployment config
+export VSPHERE_DATACENTER="SDDC-Datacenter"   # the vSphere datacenter to deploy the management cluster on
+export VSPHERE_DATASTORE="DefaultDatastore"   # the vSphere datastore to deploy the management cluster on
+export VSPHERE_NETWORK="vm-network-1"         # the VM network to deploy the management cluster on
+export VSPHERE_RESOURCE_POOL="*/Resources/Compute-ResourcePool/CAPV"  # the vSphere resource pool for your VMs
+export VSPHERE_FOLDER="Workloads"             # the VM folder for your VMs, this can be "" if not applicable to your environment
+export VSPHERE_TEMPLATE="ubuntu-16.04-server-cloudimg-amd64"  # the VM template to use for your management cluster
```

Comment:
Hi @andrewsykim,

The template should be a template a user deployed from one of the OVAs in the GCS bucket `gs://capv-images/release`. For example, in our VMC test environment the template name is `ubuntu-1804-kube-v1.13.6`. A user can learn more about the machine images in the [image documentation](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/tree/master/build/images) or list the available machine images with `gsutil`:

```shell
$ gsutil ls gs://capv-images/release/*
gs://capv-images/release/v1.13.6/:
gs://capv-images/release/v1.13.6/centos-7-kube-v1.13.6.ova
gs://capv-images/release/v1.13.6/centos-7-kube-v1.13.6.ova.sha256
gs://capv-images/release/v1.13.6/ubuntu-1804-kube-v1.13.6.ova
gs://capv-images/release/v1.13.6/ubuntu-1804-kube-v1.13.6.ova.sha256

gs://capv-images/release/v1.13.7/:
gs://capv-images/release/v1.13.7/centos-7-kube-v1.13.7.ova
gs://capv-images/release/v1.13.7/centos-7-kube-v1.13.7.ova.sha256
gs://capv-images/release/v1.13.7/ubuntu-1804-kube-v1.13.7.ova
gs://capv-images/release/v1.13.7/ubuntu-1804-kube-v1.13.7.ova.sha256

gs://capv-images/release/v1.14.0/:
gs://capv-images/release/v1.14.0/centos-7-kube-v1.14.0.ova
gs://capv-images/release/v1.14.0/centos-7-kube-v1.14.0.ova.sha256
gs://capv-images/release/v1.14.0/ubuntu-1804-kube-v1.14.0.ova
gs://capv-images/release/v1.14.0/ubuntu-1804-kube-v1.14.0.ova.sha256

gs://capv-images/release/v1.14.1/:
gs://capv-images/release/v1.14.1/centos-7-kube-v1.14.1.ova
gs://capv-images/release/v1.14.1/centos-7-kube-v1.14.1.ova.sha256
gs://capv-images/release/v1.14.1/ubuntu-1804-kube-v1.14.1.ova
gs://capv-images/release/v1.14.1/ubuntu-1804-kube-v1.14.1.ova.sha256

gs://capv-images/release/v1.14.2/:
gs://capv-images/release/v1.14.2/centos-7-kube-v1.14.2.ova
gs://capv-images/release/v1.14.2/centos-7-kube-v1.14.2.ova.sha256
gs://capv-images/release/v1.14.2/ubuntu-1804-kube-v1.14.2.ova
gs://capv-images/release/v1.14.2/ubuntu-1804-kube-v1.14.2.ova.sha256

gs://capv-images/release/v1.14.3/:
gs://capv-images/release/v1.14.3/centos-7-kube-v1.14.3.ova
gs://capv-images/release/v1.14.3/centos-7-kube-v1.14.3.ova.sha256
gs://capv-images/release/v1.14.3/ubuntu-1804-kube-v1.14.3.ova
gs://capv-images/release/v1.14.3/ubuntu-1804-kube-v1.14.3.ova.sha256
```

Reply:
Going to put to a separate doc for "How to install the CAPV OVA in vSphere" that will be a pre-req for this, thoughts? Different from the one you linked since it will just be instructions for OVA download from google storage.

Reply:
Hi @andrewsykim,

I think a doc that is top-to-bottom from how to download Kind, get CAPV machine images, install as templates, run CAPV, etc. is needed. A single walk through.

Also, please say "CAPV machine image" instead of of OVA. I am going to open a PR to completely remove the `installer` directory so there's no longer any confusion about we mean when OVA is used. Still, since they're technically "machine images" that happen to be delivered as OVAs, let's be very clear.

Reply:
Sounds good to me, I'll add a section on how to upload the CAPV machine images to vSphere.